### PR TITLE
rename c -> curl; making abort protected; minor fixes

### PR DESCRIPTION
**Pull request by pyblub**
### Type of request:

- [x] Bugfix
- [ ] Code cosmetics
- [ ] Feature enhancement
- [ ] New feature
- [ ] New plugin
- [ ] Plugin bugfix/enhancement

### Short description:

curl/chunk.py rename self.c to self.curl and minor fixes
curl/download.py add implementation for abstract method. removed line self.info.remove as the file has been moved in the previous line
curl/request.py renaming of self.c to self.curl also fix bytes/string problems

### References to this change (related pull requests, issues, etc.):

needs https://github.com/pyload/pyload/pull/3013 to be merged as well

### CODE REVIEW

File: pyload/requests/curl/chunk.py [hunk 1]
```diff
@@ -145,7 +147,7 @@ def write_header(self, buf):
     def write_body(self, buf):
         # ignore BOM, it confuses unrar
         if self.check_bom:
-            if [ord(b) for b in buf[:3]] == [239, 187, 191]:
+            if [b for b in buf[:3]] == [239, 187, 191]:
```
Comment (GammaC0de) [inline pyload/requests/curl/chunk.py]: Comparing a byte with a number does not make sense...

Comment (pyblub) [reply pyload/requests/curl/chunk.py]: Apparently, it only works in python 3. python 3 seems to convert single bytes to integers.

**Comment by GammaC0de** [comment]
@pyblub You could use:
```
from codecs import BOM_UTF8
```
Much more readable.

**Comment by pyblub** [comment]
@GammaC0de now you are telling me :smile:

Changed. good to know that this exists.

**Comment by pyblub** [comment]
anyone, any idea how long it takes to get a report from codacy/pr?

**Comment by vuolter** [comment]
@pyblub it was a bug with the codacy webhook. Should be fine now, I've manually merged it all.

**Comment by pyblub** [comment]
@vuolter thx

**Pull request merged**